catalog/app/components/JsonEditor/State.spec.js

```diff
@@ -0,0 +1,289 @@
+import {
+  iterateJsonDict,
+  iterateSchema,
+  mergeSchemaAndObjRootKeys,
+  validateOnSchema,
+} from './State'
+
+import * as booleansNulls from './mocks/booleans-nulls'
+import * as compound from './mocks/compound'
+import * as deeplyNestedArray from './mocks/deeply-nested-array'
+import * as deeplyNestedObject from './mocks/deeply-nested-object'
+import * as incorrect from './mocks/incorrect'
+import * as regular from './mocks/regular'
+
+describe('Root keys for JSON editor', () => {
```

Comment:
pls see a note on structuring test cases in a neighbour PR (#1922)

Reply:
It has this structure already. It can has some syntax or grammar mistakes, but the structure is `describe("Subject", it("verb + attributes"))`